### ISSUE TITLE
No wheels for Python3.13 on aarch64

### ISSUE THREAD

**Issue by charris** [numpy/numpy#27107]
Just a note, the manylinux and musl versions are missing. Is there a reason for this, or is it just an oversight?

**Comment by andyfaff** [comment]
An oversight. Can add them today.

**Comment by andyfaff** [comment]
(did we want free threaded as well?9

**Comment by charris** [comment]
> did we want free threaded as well?

I'm guessing yes.  We will probably want that for windows as well when it is working. @ngoldbaum Is there any free threaded progress on windows?

**Comment by ngoldbaum** [comment]
Is this aarch64 Linux? Assuming the tests pass it should be fine.

Windows needs to wait on meson getting updated.

**Comment by andyfaff** [comment]
Unfortunately the `cp313t-manylinux2014_aarch64` entry has build problems, https://cirrus-ci.com/task/5056605786472448.
@ngoldbaum 

```
[296/325] Compiling C object numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o
FAILED: numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o 
cc -Inumpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p -Inumpy/random -I../numpy/random -I../numpy/random/src -Inumpy/_core -I../numpy/_core -Inumpy/_core/include -I../numpy/_core/include -I../numpy/_core/src/common -Inumpy -I/opt/_internal/cpython-3.13.0rc1-nogil/include/python3.13t -I/project/build/meson_cpu -fvisibility=hidden -fdiagnostics-color=always -DNDEBUG -Wall -Winvalid-pch -std=c11 -O3 -fno-strict-aliasing -DNPY_HAVE_NEON_VFPV4 -DNPY_HAVE_NEON_FP16 -DNPY_HAVE_NEON -DNPY_HAVE_ASIMD -fPIC -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -DNPY_NO_DEPRECATED_API=0 -MD -MQ numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o -MF numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o.d -o numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o -c numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:2380:80: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
 2380 | static CYTHON_INLINE PyObject *__Pyx_PyVectorcall_FastCallDict(PyObject *func, __pyx_vectorcallfunc vc, PyObject *const *args, size_t nargs, PyObject *kw);
      |                                                                                ^~~~~~~~~~~~~~~~~~~~
      |                                                                                vectorcallfunc
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11238:69: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
11238 | static PyObject *__Pyx_PyVectorcall_FastCallDict_kw(PyObject *func, __pyx_vectorcallfunc vc, PyObject *const *args, size_t nargs, PyObject *kw)
      |                                                                     ^~~~~~~~~~~~~~~~~~~~
      |                                                                     vectorcallfunc
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11283:80: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
11283 | static CYTHON_INLINE PyObject *__Pyx_PyVectorcall_FastCallDict(PyObject *func, __pyx_vectorcallfunc vc, PyObject *const *args, size_t nargs, PyObject *kw)
      |                                                                                ^~~~~~~~~~~~~~~~~~~~
      |                                                                                vectorcallfunc
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c: In function ‘__Pyx_CyFunction_CallAsMethod’:
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11972:6: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
11972 |      __pyx_vectorcallfunc vc = __Pyx_CyFunction_func_vectorcall(cyfunc);
      |      ^~~~~~~~~~~~~~~~~~~~
      |      vectorcallfunc
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:2470:45: warning: initialization of ‘int’ from ‘vectorcallfunc’ {aka ‘struct _object * (*)(struct _object *, struct _object * const*, long unsigned int,  struct _object *)’} makes integer from pointer without a cast [-Wint-conversion]
 2470 | #define __Pyx_CyFunction_func_vectorcall(f) (((PyCFunctionObject*)f)->vectorcall)
      |                                             ^
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11972:32: note: in expansion of macro ‘__Pyx_CyFunction_func_vectorcall’
11972 |      __pyx_vectorcallfunc vc = __Pyx_CyFunction_func_vectorcall(cyfunc);
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11975:16: warning: implicit declaration of function ‘__Pyx_PyVectorcall_FastCallDict’; did you mean ‘__Pyx_PyObject_FastCallDict’? [-Wimplicit-function-declaration]
11975 |         return __Pyx_PyVectorcall_FastCallDict(func, vc, &PyTuple_GET_ITEM(args, 0), (size_t)PyTuple_GET_SIZE(args), kw);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                __Pyx_PyObject_FastCallDict
numpy/random/_mt19937.cpython-313t-aarch64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11975:16: warning: returning ‘int’ from a function with return type ‘PyObject *’ {aka ‘struct _object *’} makes pointer from integer without a cast [-Wint-conversion]
11975 |         return __Pyx_PyVectorcall_FastCallDict(func, vc, &PyTuple_GET_ITEM(args, 0), (size_t)PyTuple_GET_SIZE(args), kw);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by charris** [comment]
Looks like a cython version problem.

**Comment by andyfaff** [comment]
The cython install is:
```
+ python -m pip install -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython
Looking in indexes: https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
Collecting cython
  Downloading https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/cython/3.1.0a0/Cython-3.1.0a0-py3-none-any.whl (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 13.6 MB/s eta 0:00:00
Installing collected packages: cython
Successfully installed cython-3.1.0a0
```

It's installing from nightly because of:
https://github.com/numpy/numpy/blob/main/tools/wheels/cibw_before_build.sh#L62

**Comment by EwoutH** [comment]
Looks like quite some fixes have been made on the cython [master](https://github.com/cython/cython/commits/master/) branch in the last few hours. It might be worth re-triggering the CI builds and see if that resolves the problem.

**Comment by charris** [comment]
@EwoutH It has been busy indeed, but the nightly version hasn't updated yet.

**Comment by charris** [comment]
I think the way forward is to ignore the problem for now and see if it gets fixed in the next couple of weeks. We should probably report it upstream, we might learn if it is fixed.

**Comment by andyfaff** [comment]
I did report, but not in it's own issue

**Comment by ngoldbaum** [comment]
I'll take a look at this as well sometime today.

**Comment by EwoutH** [comment]
3.0.11 also just got tagged: https://github.com/cython/cython/releases/tag/3.0.11

**Comment by ngoldbaum** [comment]
I just set up a numpy dev environment on an aarch64 Linux VM image. I'm able to build NumPy on the VM image with a free-threaded python so I suspect the cython nightly isn't properly getting installed in the cibuildwheel environment given the error you're seeing.

@EwoutH that release won't support free-threading, for what it's worth. 3.1.0 will be the first released version to support free-threading.

**Comment by ngoldbaum** [comment]
The "full" tests pass on the VM image, so hopefully if you make sure cibuildwheel uses no-build-isolation it will just work.

**Comment by charris** [comment]
Fixed by #27110.
